test/e2e/harness.go

```diff
@@ -70,56 +70,51 @@ func newAppHarness() *harness {
+		return nil, errors.Wrap(err, "failed to deploy native contract")
 	}
 
-	txStatus, executionResult := sendTxOut.TransactionStatus, sendTxOut.ExecutionResult
-
-	for txStatus == codec.TRANSACTION_STATUS_PENDING {
+	for txOut.TransactionStatus == codec.TRANSACTION_STATUS_PENDING {
 		// check timeout
 		if time.Now().Sub(beginTime) > timeoutDuration {
-			return "", "", fmt.Errorf("contract deployment is TRANSACTION_STATUS_PENDING for over %v", timeoutDuration)
+			return nil, fmt.Errorf("contract deployment is TRANSACTION_STATUS_PENDING for over %v", timeoutDuration)
 		}
 
 		time.Sleep(10 * time.Millisecond)
 
-		txStatusOut, _ := h.getTransactionStatus(txId)
-
-		txStatus, executionResult = txStatusOut.TransactionStatus, txStatusOut.ExecutionResult
+		txOut, _ = h.getTransactionStatus(txId)
 	}
 
-	return executionResult, txStatus, err
+	return txOut, err
 }
 
-func (h *harness) sendTransaction(senderPublicKey []byte, senderPrivateKey []byte, contractName string, methodName string, args ...interface{}) (response *codec.SendTransactionResponse, txId string, err error) {
+func (h *harness) sendTransaction(senderPublicKey []byte, senderPrivateKey []byte, contractName string, methodName string, args ...interface{}) (*codec.TransactionResponse, string, error) {
 	payload, txId, err := h.client.CreateTransaction(senderPublicKey, senderPrivateKey, contractName, methodName, args...)
 	if err != nil {
 		return nil, txId, err
 	}
-	response, err = h.client.SendTransaction(payload)
-	return
+	out, err := h.client.SendTransaction(payload)
+	return out.TransactionResponse, txId, err
 }
 
-func (h *harness) sendDeployTransaction(senderPublicKey []byte, senderPrivateKey []byte, contractName string, code ...[]byte) (response *codec.SendTransactionResponse, txId string, err error) {
+func (h *harness) sendDeployTransaction(senderPublicKey []byte, senderPrivateKey []byte, contractName string, code ...[]byte) (*codec.TransactionResponse, string, error) {
 	payload, txId, err := h.client.CreateDeployTransaction(senderPublicKey, senderPrivateKey, contractName, orbsClient.PROCESSOR_TYPE_NATIVE, code...)
 	if err != nil {
 		return nil, txId, err
 	}
-	response, err = h.client.SendTransaction(payload)
-	return
+	out, err := h.client.SendTransaction(payload)
+	return out.TransactionResponse, txId, err
 }
 
-func (h *harness) eventuallyRunQueryWithoutError(timeout time.Duration, senderPublicKey []byte, contractName string, methodName string, args ...interface{}) (response *codec.RunQueryResponse, err error) {
+func (h *harness) runQueryAtBlockHeight(timeout time.Duration, expectedBlockHeight uint64, senderPublicKey []byte, contractName string, methodName string, args ...interface{}) (response *codec.RunQueryResponse, err error) {
 	test.Eventually(timeout, func() bool {
 		response, err = h.runQuery(senderPublicKey, contractName, methodName, args...)
-		return err == nil
+		return err == nil && response.BlockHeight >= expectedBlockHeight
 	})
 	return response, err
```

Comment:
If the height was never reached then this would return with a response for a lower height

Reply:
no, that's the whole point here

Reply:
actually yes